### PR TITLE
Task requests migrations scripts

### PR DESCRIPTION
**Pull request by Ajeyakrishna-k**
Date: 4 November 2023
Developer Name: @Ajeyakrishna-k 

----

## Issue Ticket Number:- 
-  #1613

## Description: 
Adds scripts to migrate task requests with old document schema to the new schema. The schema was changed to implement task creation requests. These are one time scripts and will be removed after their application

Is Under Feature Flag 
- [ ] Yes
- [x] No

Database changes
- [ ] Yes
- [x] No

Breaking changes (If your feature is breaking/missing something please mention pending tickets)
- [ ] Yes
- [x] No

### Is Development Tested?

- [x] Yes
- [ ] No

Test code coverage: 
Screenshot 2023-11-04 at 2 44 46 AM

### CODE REVIEW

File: models/taskRequests.js [hunk 1]
```diff
@@ -338,10 +338,74 @@ const approveTaskRequest = async (taskRequestId, user) => {
...
+          }
+
+          return {
+            userId: requestorId,
+            status: userStatus,
+          };
+        });
+        const updatedTaskRequestData = {
+          ...taskRequestData,
+          requestType: TASK_REQUEST_TYPE.ASSIGNMENT,
```
Comment (heyrandhir) [inline models/taskRequests.js:367]: can the old taskreq not be task creation as we are assigning assignment by default.

**Review by heyrandhir — APPROVED** [review]
Good work @Ajeyakrishna-k . Just a small query if you can ans that

File: routes/taskRequests.js [hunk 2]
```diff
@@ -12,4 +12,6 @@ router.get("/:id", authenticate, authorizeRoles([SUPERUSER]), taskRequests.fetch
 router.post("/addOrUpdate", authenticate, validateUser, taskRequests.addOrUpdate);
 router.patch("/approve", authenticate, authorizeRoles([SUPERUSER]), validateUser, taskRequests.approveTaskRequest);
 router.post("/", authenticate, validators.postTaskRequests, taskRequests.addTaskRequests);
+router.post("/migrations", authenticate, authorizeRoles([SUPERUSER]), taskRequests.migrateTaskRequests);
```
Comment (heyrandhir) [inline routes/taskRequests.js:15]: `migrations` isn't a noun. can you please update this.

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: routes/taskRequests.js (hunk 2 above)
Comment (Ajeyakrishna-k) [reply routes/taskRequests.js:15]: - `migrate` is a verb and `migrations` is a noun. 
- Other names I can think about are `/scripts` or `/migration-scripts`. Let me know what you suggest
Also note these are a one time script

**Review by heyrandhir — APPROVED** [review]

**Review by iamitprakash — COMMENTED** [review]
why in database changes yes and No both are marked

**Comment by Ajeyakrishna-k** [comment]
> why in database changes yes and No both are marked

Updated.

**Review by ajoykumardas12 — APPROVED** [review]

**Pull request merged**